File: CSharpCodeBase/components/controlsystem/enemyactions/actionhit.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class actionhit :action{
 public void init (self, entity);
       Action.init(self, 4, false, "hit");
       this.entity = entity;
 }})
 public void OnStartRunning(){
   //Action.OnStart(self)
   this.entity.mover:Stop();
   this.entity.mecanim:SetFloat("action_type", 0);
   this.entity.mecanim:ForceSetState("Hit");
   this.isHitStarted = false;
 }
 public void Update(){
   if(not this.isHitStarted  ){ 
     this.isHitStarted = this.entity.mecanim:CheckStateName("Hit");
   }
   return this.isHitStarted  &&  not this.entity.mecanim:CheckStateName("Hit");
 }
 }}

File: CSharpCodeBase/components/controlsystem/enemyactions/actionhit.lua
local Action = require 'components.controlsystem.action'
local Class = require 'utils.hump.class'

local ActionHit = Class({__includes=Action, init = function (self, entity)
      Action.init(self, 4, false, 'hit')
      self.entity = entity
end})

function ActionHit:OnStartRunning()
  --Action.OnStart(self)
  self.entity.mover:Stop()
  self.entity.mecanim:SetFloat("action_type", 0)
  self.entity.mecanim:ForceSetState("Hit")
  self.isHitStarted = false
end

function ActionHit:Update()
  if not self.isHitStarted then
    self.isHitStarted = self.entity.mecanim:CheckStateName("Hit")
  end
  return self.isHitStarted and not self.entity.mecanim:CheckStateName("Hit")
end

return ActionHit